packages/notify-client/src/controllers/engine.ts

```diff
@@ -1218,6 +1244,29 @@ export class NotifyEngine extends INotifyEngine {
     );
   };
 
+  // returns true if statement is stale, false otherwise.
+  private checkIfSignedStatementIsStale = (
+    account: string,
+    currentStatement: string
+  ) => {
+    const hasSignedStatement =
+      this.client.signedStatements.keys.includes(account);
+    if (!hasSignedStatement) {
+      // if there is no signed statement, then this account's statement was signed
+      // previous to this function (and thus the latest statement) being introduced
+      // therefore, it is stale.
+      return true;
+    }
+
+    const signedStatement = this.client.signedStatements.get(account);
+
+    if (signedStatement.statement !== currentStatement) {
```

Comment:
Can simplify to the bool returned by the condition itself, also avoids the explicit `return false` at the end:

```suggestion
return signedStatement.statement !== currentStatement
```